src/components/inputs/radio-input/radio-group.js

```diff
@@ -36,25 +52,49 @@ class Group extends React.PureComponent {
   }
 
   render() {
-    const DirectionWrapper =
-      this.props.direction === 'stack' ? Spacings.Stack : Spacings.Inline;
+    const optionElements = React.Children.map(
+      this.props.children,
+      (child, index) => {
+        // NOTE: Allowing to intersperse other elements than `Option`.
+        if (child && child.type.displayName === Option.displayName) {
+          return React.cloneElement(child, {
+            id: `${this.props.id}-${index}`,
+            name: this.props.name,
+            isChecked: this.props.value === child.props.value,
+            isDisabled: child.props.isDisabled || this.props.isDisabled,
+            isReadOnly: this.props.isReadOnly,
```

Comment:
I find this functionality confusing. Maybe the prop names are wrong.

isDisabled makes every single radio option disabled?